pkg/rancher-components/src/components/RcDropdown/RcDropdownItemCheckbox.vue

```diff
@@ -0,0 +1,138 @@
+<script setup lang="ts">
+/**
+ * An item for a dropdown menu. Used in conjunction with RcDropdown.
+ */
+import { useDropdownItem } from '@components/RcDropdown/useDropdownItem';
+import { generateRandomAlphaString } from '@shell/utils/string';
+
+const props = defineProps({ modelValue: Boolean, disabled: Boolean });
+const emits = defineEmits(['click']);
+
+const { handleKeydown, handleActivate } = useDropdownItem();
+
+const handleClick = () => {
+  if (props.disabled) {
+    return;
+  }
+
+  emits('click', !props.modelValue);
+};
+
+const uid = `menu-item-checkbox-${ generateRandomAlphaString(10) }`;
+const uidLabel = `menu-item-checkbox-label-${ generateRandomAlphaString(10) }`;
+</script>
+
+<template>
+  <div
+    ref="dropdownMenuItem"
+    dropdown-menu-item
+    tabindex="-1"
+    role="menuitemcheckbox"
+    :disabled="disabled || null"
+    :aria-disabled="disabled || false"
+    @click.stop="handleClick"
+    @keydown.enter.space="handleActivate"
+    @keydown.up.down.prevent.stop="handleKeydown"
+  >
+    <div class="rc-checkbox-container">
+      <input
+        :id="uid"
+        class="rc-checkbox"
+        role="checkbox"
+        type="checkbox"
+        :checked="modelValue"
+        :aria-checked="modelValue"
+        :aria-describedby="uidLabel"
+      >
+      <label
+        :id="uidLabel"
+        class="rc-checkbox-label"
+      >
+        <slot name="default">
+          <!--Empty slot content-->
+        </slot>
+      </label>
```

Comment:
@aalves08 We are using a custom checkbox implementation here because the existing Checkbox component is not centered properly. I attempted to update the styles to resolve this issue, but I still couldn't get it quite right. There's something about our checkbox component that I'm missing.. 

I'm open to recommendations on how we can fix the checkbox component. Otherwise, if we like this, we might have a path forward to simplify the existing checkbox implementation. 

@edenhernandez-suse

Reply:
Is it that bad @rak-phillip ? What's the visual difference here?

Reply:
It's not horrible, but the existing checkbox component is decidedly off-center:

![image](https://github.com/user-attachments/assets/5a64e3bb-362e-4e8f-b52a-a4319a47775e)

I'm willing to attempt another go-around at trying to fix this.. although, I'm still not sure what the issue is.